File: src/main/java/com/github/thorben/webnd/WebNDApplication.java
package com.github.thorben.webnd;

import com.github.thorben.webnd.application.SessionStorage;
import com.github.thorben.webnd.application.tech.MessageBroker;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Scope;
import org.springframework.context.annotation.ScopedProxyMode;
import org.springframework.web.context.WebApplicationContext;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;

@SpringBootApplication
public class WebNDApplication {

	public static void main(String[] args) {
		SpringApplication.run(WebNDApplication.class, args);
	}

	@Bean
	@Scope(value = WebApplicationContext.SCOPE_SESSION, proxyMode = ScopedProxyMode.TARGET_CLASS)
	public MessageBroker messageBroker() {
		return new MessageBroker();
	}

	@Bean
	@Scope(value = WebApplicationContext.SCOPE_SESSION, proxyMode = ScopedProxyMode.TARGET_CLASS)
	public SessionStorage sessionStorage() {
		return new SessionStorage();
	}

}

File: src/main/kotlin/com/github/thorben/webnd/WebNDApplication.kt
package com.github.thorben.webnd

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.KotlinModule
import com.github.thorben.webnd.application.SessionStorage
import com.github.thorben.webnd.domain.character.AttributeTypeRepository
import com.github.thorben.webnd.domain.character.BaseValueTypeRepository
import com.github.thorben.webnd.domain.character.DetailTypeRepository
import com.github.thorben.webnd.domain.character.SkillTypeRepository
import com.github.thorben.webnd.domain.games.GameTypeRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.context.annotation.SessionScope

fun main(args: Array<String>) {
	SpringApplication.run(WebNDApplication::class.java, *args)
}

@SpringBootApplication
@Configuration
class WebNDApplication {

	@Bean
	fun mapper(): ObjectMapper {
		return ObjectMapper().registerModule(KotlinModule())
	}
}